### PR TITLE
[TMVA] JsMVA -- Update d3 lib

### PR DESCRIPTION
**Pull request by ashlaban**
The name of the d3 library included with jsroot was changed from d3.v3
to just d3. This updates JsMVA to use the new name.

Note: Not sure if more changes are required to propagate to SWAN and notebook environment. Please advise :)

### CODE REVIEW

File: etc/notebook/JsMVA/js/JsMVA.min.js [hunk 1]
```diff
@@ -1 +1 @@
...
....attr("x",canvas.width-s.pos.x+s.rect.width+s.padding).attr("y",s.pos.y+r*s.dy+s.rect.height).text(function(t){return t.fTitle}).style("fill",function(t){return e.Painter.root_colors[t.fFillColor]})});o.append("svg").attr("width","55px").attr("height","20px").style({position:"absolute",bottom:"15px",right:"40px"}).append("text").attr("x","5px").attr("y","15px").text(r.fGraphs.arr[0].fTitle.indexOf("Error on training set")!=-1?"Epoch":"#tree").style({"font-size":"16px"})})};t.drawTrainingTestingErrors=function(t,i){var n=e.parse(i);e.draw(t,n);r(t,n)};t.updateTrainingTestingErrors=function(t,i){var n=e.parse(i);e.redraw(t,n);r(t,n)};t.NetworkDesigner=function(t,e){require(["NetworkDesigner"],function(e){e.draw(t)})};t.outputShowCorrelationMatrix=function(t){require(["jquery","jquery-ui"],function(r){var i=e.parse(r("#"+t).html());if(!r("#dialog_"+t).length||r("#dialog_"+t).length<1){r("#"+t).parent().append("<div id='dialog_"+t+"' title='"+i.fTitle+"' style='width: 600px; height: 340px; z-index: 99;'></div>");e.draw("dialog_"+t,i,"colz;PAL50;text")}r("#dialog_"+t).dialog({autoOpen:true,width:600,show:{effect:"blind",duration:1e3},hide:{effect:"explode",duration:500}})})};return t});
```
Comment (ashlaban) [inline etc/notebook/JsMVA/js/JsMVA.min.js]: The only change here is `require.config({paths:{d3:e+"d3.v3.min" ...` to `equire.config({paths:{d3:e+"d3.min"`.

Comment (etejedor) [reply etc/notebook/JsMVA/js/JsMVA.min.js]: I understand this makes it work again in a notebook? Did you try locally in your computer with a local Jupyter server?

Comment (ashlaban) [reply etc/notebook/JsMVA/js/JsMVA.min.js]: I tried, and verifed, it with modifying the request that my local client does to the server (using the chrome plugin Tamper I changed the request from d3.v3.min.js to d3.min.js).

However, I can not test this using my local environment as jsMVA.min.js is fetched from `https://root.cern.ch/js/jsmva/latest/js`, not my local root bin/src dir. (I also tried changing pyroot to fetch jsmva from a local webserver, but I was not able to make this work).

Comment (etejedor) [reply etc/notebook/JsMVA/js/JsMVA.min.js]: To be completely sure this fixes it, perhaps @bellenot can replace `jsMVA.min.js` by your fixed version on root.cern.ch. Anyway, now it is broken.

Comment (bellenot) [reply etc/notebook/JsMVA/js/JsMVA.min.js]: Sorry, but I don't see where d3 was changed from d3.v3.min.js to d3.min.js...

Comment (ashlaban) [reply etc/notebook/JsMVA/js/JsMVA.min.js]: It should suffice to replace any occurrence of "d3.v3.min" with "d3.min" in the file on the web server.

Comment (bellenot) [reply etc/notebook/JsMVA/js/JsMVA.min.js]: I just want to understand what has changed and where. I can change whatever you want in jsmva (not in jsroot), but I want to know who is in charge of those files...

Comment (bellenot) [reply etc/notebook/JsMVA/js/JsMVA.min.js]: OK, I created a `jsmva/1.1` directory with the required changes, and the `jsmva/latest` symlink is now pointing to it. Please let me know if it solves the issue.

**Comment by ashlaban** [comment]
Thanks!

File: etc/notebook/JsMVA/js/JsMVA.min.js (hunk 1 above)
Comment (ashlaban) [reply etc/notebook/JsMVA/js/JsMVA.min.js]: Thanks @bellenot!

From my testing it seems to work now. Waiting to hear back from the reporter here: https://root-forum.cern.ch/t/tmva-not-showing-plots-on-swan/29762/6

**Comment by bellenot** [comment]
BTW, I replaced all occurrences of "d3.v3.min" with "d3.min" (not only in JsMVA.js and JsMVA.min.js). So you might want to change other files as well...

**Comment by ashlaban** [comment]
Thanks for pointing out the other locations @bellenot. Fixed now.

**Comment by ellert** [comment]
This PR duplicates one of the commits in PR #1520 (that PR in addition addresses two more issues in JsMVA not related to the d3 change)

As written in the description of PR #1520:

> The code has not been updated to reflect changes in jsroot. The code still tries to load d3.v3.min.js from jsroot. In the current version of jsroot the d3 script has been updated to version 4 and the file is now called d3.min.js. The pull request changes the name of the file in the two places where it is referenced, so it should now find the file. However this change is incomplete since the d3 API changes between the versions and some porting is needed.

This PR does not introduce any other changes than changing the name of the js file in the two places where it is referenced in the code, just as PR #1520 does. The API of d3 v4 is not fully backward compatible with d3 v3. Did you go through the code and check that nothing else needs to be changed? In jsroot itself there were many other changes to the code to adapt to the changed API when the d3.js file was updated to v4 as seen in the commit:

https://github.com/root-project/jsroot/commit/c156b8956f77d1aef4b70cd8609c4b1f2036f94f#diff-26681b69718507c05077030d69e5ba58

One more comment: In addition to the change that changes the name of the d3.js file, this PR also proposes changes to some error messages in jsroot. These proposed changes should be proposed to jsroot upstream (https://github.com/root-project/jsroot), otherwise they will be overwritten next time root's bundled copy of jsroot is updated by copying files from upstream.

**Comment by ashlaban** [comment]
Thanks for your comments @ellert. I'm going through the how d3.v3 -> d3.v4 affects jsMVA rights now and will update your PR with the results. This one I will close.

**Pull request closed without merging**